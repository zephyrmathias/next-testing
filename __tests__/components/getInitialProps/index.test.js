import React from 'react'
import { shallow } from 'enzyme'
import Example from '../../../components/getInitialProps'

describe('snapshot', () => {
  it('should match with snapshot', () => {
    const component = shallow(<Example />)
    expect(component).toMatchSnapshot()
  })
})

/**
 * getInitialProps is static class method
 * so we need to execute it by calling ClassName.getInitialProps()
 */
describe('getInitialProps', () => {
  it('should return data props as "server"', () => {
    const expectedResult = { data: 'server' }
    const res = Example.getInitialProps({ isServer: true })

    // toStricEqual is deep comparison
    expect(res).toStrictEqual(expectedResult)
  })

  it('should return data props as "client"', () => {
    const expectedResult = { data: 'client' }
    const res = Example.getInitialProps({ isServer: false })
    expect(res).toStrictEqual(expectedResult)
  })
})
