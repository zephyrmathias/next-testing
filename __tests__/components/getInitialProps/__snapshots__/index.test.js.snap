// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should match with snapshot 1`] = `
<div>
  Hello world.
</div>
`;
